pkg/artifactory/provider.go

```diff
@@ -232,3 +245,50 @@ func buildClient(d *schema.ResourceData) (*http.Client, auth2.ServiceDetails, er
 		return nil, nil, fmt.Errorf("either [username, password] or [api_key] or [access_token] must be set to use provider")
 	}
 }
+
+func createJfrogXrayClient(d *schema.ResourceData, xrayURL string) (*jfrogxray.XrayServicesManager, error) {
+	details := jfrogxrayauth.NewXrayDetails()
+
+	username := d.Get("username").(string)
+	password := d.Get("password").(string)
+	apiKey := d.Get("api_key").(string)
+
+	if len(xrayURL) <= 0 {
```

Comment:
Basically, what I did in this file was allow URL to break it into parts (and thus remove extra `/`) and then reassembled it properly:

```go
	u, err := url.ParseRequestURI(d.Get("url").(string))

	if err != nil {
		return nil, err
	}
	baseUrl := fmt.Sprintf("%s://%s", u.Scheme, u.Host)
	artifactoryEndpoint := fmt.Sprintf("%s/artifactory/", baseUrl)
```

Reply:
Hi @chb0github,
I left this code in the function to protect it from any call to this function.

Did you want me to use a similar pattern in the `createJfrogXrayClient` function, and require base url, instead of the xray url?

Reply:
We have to use the base url because we (as a company) intend to fold in all of our products.